js/foundation.magellan.js

```diff
@@ -100,11 +100,15 @@ class Magellan extends Plugin {
         e.preventDefault();
         var arrival   = this.getAttribute('href');
         _this.scrollToLoc(arrival);
+        _this.calcPoints();
+        _this._updateActive();
```

Comment:
I investigated on this issue before. The easy solution is to put `this.calcPoints();` everywhere like @kball did 1 year ago (https://github.com/zurb/foundation-sites/pull/9491), and according to me this should work.

But `calcPoints()` is a huge operation that will get the position of all targets on the page. When the plugin was implemented, `calcPoints()` was only called on init/reflow so `_updateActive` on the scroll event works on pre-calculated positions. Calling `calcPoints` everywhere is against this optimized pattern.

So I think we should rather do the following:
* Consider a "page is loaded" when we receive the `onload` event OR the page is already loaded when the plugin is instantiated (related to https://github.com/zurb/foundation-sites/pull/10947).
* Wait for the page to load to listen for scroll events (or others events that trigger `_updateActive`).
  If the user begin to scroll before, we cannot update the magellan anchors anyway.
* Wait for the page to load before scrolling in tests.

Reply:
Mh, sounds like a small refactor for this. Should we handle it in this PR or another one?

Reply:
I think we should clean that in this PR, as this is "anti-pattern" and can introduce some performance issues.

Reply:
Could need some advice. Do you have some specific solution in your mind how this should look like?

Reply:
hmm... For example
1. An an utility to check for `readyState === "complete"`, and will create a listener on `$(window).once('load', ...)` otherwise.
2. Use it in the `_event` function and start listening all others events in it (so we listen nothing before the page is loaded).
3. For the tests, I guess we can still use something like `$html.once('load',  ...)` and call `plugin.scrollToLoc(...)` within.